doc/user/content/guides/node-js.md

```diff
@@ -0,0 +1,245 @@
+    ssl: {
+        ca   : fs.readFileSync("server-ca.pem").toString(),
+        key  : fs.readFileSync("client-key.pem").toString(),
+        cert : fs.readFileSync("client-cert.pem").toString(),
+    }
+});
+
+async function main() {
+    await client.connect();
+    /* Work with Materialize */
+}
+
+main();
+```
+
+Replace `MY_INSTANCE_ID`) in the code above with your Materialize Cloud instance ID. The [`ssl` property](https://node-postgres.com/features/ssl), loads the certificate files downloaded from the Materialize Cloud [Connect](/cloud/connect-to-materialize-cloud/) dialog.
+
+## Stream
+
+To take advantage of incrementally updated materialized views from a Node.js application, instead of [querying](#query) Materialize for the state of a view at a point in time, use [a `TAIL` statement](/sql/tail/) to request a stream of updates as the view changes.
+
+Read a stream of updates from a materialized view named `my_view` into Node.js:
+
+```js
+const { Client } = require('pg');
+
+async function main() {
+  const client = new Client('postgres://materialize@localhost:6875/materialize');
+  await client.connect();
+
+  await client.query('BEGIN');
+  await client.query('DECLARE c CURSOR FOR TAIL my_view');
+
+  while (true) {
+    const res = await client.query('FETCH ALL c');
+    console.log(res.rows);
+  }
+}
+
+main();
+```
+
+The code above opens a long-lived transaction with `BEGIN` and uses [`TAIL` with `FETCH`](/sql/tail/#tailing-with-fetch) to receive a stream of updates to `my_view`. The [TAIL Output format](/sql/tail/#output) of `res.rows` is an array of view update objects.
+
+**Note on output format:** When a row of a tailed view is **updated,** two objects will show up in the `rows` array:
+
+```js
+[
+    ...
+    {
+        mz_timestamp: '1627225629000',
+        mz_diff: '1',
+        my_column_one: 'ABC',
+        my_column_two: 'new_value'
+    },
+    {
+        mz_timestamp: '1627225629000',
+        mz_diff: '-1',
+        my_column_one: 'ABC',
+        my_column_two: 'old_value'
+    },
+    ...
+]
+```
+
+An `mz_diff` value of `-1` indicates Materialize is deleting one row with the included values.  An update is just a deletion (`mz_diff: '-1'`) and an insertion (`mz_diff: '1'`) with the same `mz_timestamp`.
+
+<!--- NOT PUBLISHABLE UNTIL https://github.com/brianc/node-postgres/pull/2573 is merged
+### Using pg-query-stream
+
+```js
+const { Client } = require('pg');
+const QueryStream = require('pg-query-stream');
+
+const client = new Client('postgres://materialize@localhost:6875/materialize');
+
+client.connect((err, client) => {
+  if (err) {
+    throw err;
+  }
+  const stream = client.query(new QueryStream('TAIL avg_bid', []));
+  stream.pipe(process.stdout);
+});
+```
+--->
+
+## Query
+
+Querying Materialize is identical to querying a traditional PostgreSQL database: Node.js executes the query, and Materialize returns the state of the view, source, or table at that point in time.
+
```

Comment:
This is the view we created with TAIL in the previous section?

Reply:
I think this is where naming this a "cheatsheet" is probably clearer in the long term, it is not building sequentially it is just demonstrating how to do all the common operations from a node.js app.